### DISCUSSION TITLE
Upper Limit for Battery Capacity?

### DISCUSSION THREAD

**Discussion by TechFreak9356** [PycmShoma/IncreasedBatteryCapacity#2] [Q&A]
Hi there. I have a personalized Mi 9T project with active cooling and extra batteries, and someone has suggested to me that I try your module to fix the battery capacity issue. I managed to get your module to work with my raphael, and modify the service.sh and power_profile.xml so that my extra 4000mAh battery is recognized. However, since my project calls for 2 extra batteries, that's 12000mAh in total.

I was able to get your module to write 12000mAh to my Power Profile, and it actually charges up to there (though Accubattery says around 10000mAh capacity only) but I was unable to get it to correct the Kernel Profile. Somehow the kernel profile gets reverted back to 4000mAh. 

I also tried to use Scene5 app to change the full_cap value from $ bms/uevent, and it maxes out at 9999mAh (if I set it to 12000mAh, it shows 1200mAh instead) and Device Info HW reports the kernel profile is back to 4000mAh. 

What puzzles me is that if I set the value to below 10000mAh (say 9999mAh), it works. I have tested this first before my 3rd battery arrived with modifying to 8000mAh, and it works fine (both old and new versions).

I would like some help over this matter, or at least if you could tell me if the kernel profile is hardcoded on the kernel. I currently use steroid kernel v0.7.9 and the developers on tg before said that their kernel supports any battery capacity. However, I couldn't find a way to get the battery capacity to be configured properly until somebody referred to your module to me on xda.

You have made a great module indeed. It helped a lot with configuring the battery capacity properly, and I thank you in advance for any help in this regard. 

I apologize for messing around with your code if it's not allowed. I have no intention of taking any credit for anything, I just wanted my two extra batteries to work properly.

**Comment by PycmShoma** [comment]
Greetings.
Unfortunately, I do not have enough time to deal with other devices now.
It is better to contact one of the developers of the kernels for your device and consult with them

**Reply by TechFreak9356** [reply]
All right, thank you very much. I will try to consult them. Your module works fine on raphael by just changing device = cepheus to raphael on customize.sh. I was only wondering what is the upper limit of the charge_full and charge_full_design values on $ bms/uevent, and if they were kernel limited or kernel controlled. Thanks a lot
